.github/workflows/pull_request.yml

```
@@ -0,0 +1,38 @@
+name: PR
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  docker-build-push:
+    strategy:
+      matrix:
+        images: [eq-k8s-helm-deploy-image, eq-terraform-deploy-image, eq-python-deploy-image, eq-tinyproxy-image]
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
```

```suggestion
      - uses: actions/checkout@v4
```
We could go straight to v4 here